sentry-samples/sentry-samples-console/src/main/java/io/sentry/samples/console/Main.java

```diff
@@ -0,0 +1,132 @@
+          // your Sentry project/dashboard
+          options.setDsn("https://f7f320d5c3a54709be7b28e0f2ca7081@sentry.io/1808954");
+
+          // Modifications to event before it goes out. Could replace the event altogether
+          options.setBeforeSend(
+              (event, hint) -> {
+                // Drop an event altogether:
+                if (event.getTag("SomeTag") != null) {
+                  return null;
+                }
+                return event;
+              });
+
+          // Allows inspecting and modifying, returning a new or simply rejecting (returning null)
+          options.setBeforeBreadcrumb(
+              (breadcrumb, hint) -> {
+                // Don't add breadcrumbs with message containing:
+                if (breadcrumb.getMessage() != null
+                    && breadcrumb.getMessage().contains("bad breadcrumb")) {
+                  return null;
+                }
+                return breadcrumb;
+              });
+
+          // Configure the background worker which sends events to sentry:
+          // Wait up to 5 seconds before shutdown while there are events to send.
+          options.setShutdownTimeout(5000);
+
+          // Enable SDK logging with Debug level
+          options.setDebug(true);
+          // To change the verbosity, use:
+          options.setDiagnosticLevel(SentryLevel.LOG);
+
+          // Using a proxy:
+          options.setProxy(null); // new Proxy(Type.HTTP, new InetSocketAddress(8090)));
+
+          // Exclude classes from certain packages from stack traces sent to Sentry:
+          options.addInAppExclude("org.jboss");
+        });
+
+    Sentry.addBreadcrumb(
+        "A 'bad breadcrumb' that will be rejected because of 'BeforeBreadcrumb callback above.'");
+
+    // Data added to the root scope (no PushScope called up to this point)
+    // The modifications done here will affect all events sent and will propagate to child scopes.
+    Sentry.configureScope(
+        scope -> {
+          scope.addEventProcessor(new SomeEventProcessor());
+
+          scope.setExtra("SomeExtraInfo", "Some value for extra info");
+        });
+
+    // Configures a scope which is only valid within the callback
+    Sentry.withScope(
+        scope -> {
+          scope.setLevel(SentryLevel.FATAL);
+          scope.setTransaction("main");
+
+          Sentry.captureMessage("Fatal message!");
+        });
+
+    Sentry.captureMessage("Some warning!", SentryLevel.WARNING);
+
+    Exception exception = new RuntimeException("Attempting to send this multiple times");
+
+    // Only the first capture will be sent to Sentry
+    for (int i = 0; i < 3; i++) {
+      // The SDK is able to detect duplicate events:
+      // This is useful, for example, when multiple loggers log the same exception. Or exception is
+      // re-thrown and recaptured.
+      Sentry.captureException(exception);
```

Comment:
Not sure we have this de-duplication here though, need to check

Reply:
You're right, the deduplication does not happen.

Reply:
yep, it's a TODO